crash backoff in dapr operator after upgrade to 1.4.3 (helm doesn't upgrade CRDs definitions)

<!-- If you need to report a security issue please visit https://docs.dapr.io/operations/support/support-security-issues -->

## In what area(s)?

<!-- Remove the '> ' to select -->

 /area operator

## What version of Dapr?

<!-- Delete all but your choice -->

1.4.3 
## Expected Behavior
dapr operator starts correctly

## Actual Behavior
crash backoff in dapr operator after upgrade to 1.4.3. 
Related: 
- https://github.com/dapr/dapr/pull/3732
- https://github.com/dapr/dapr/issues/3731

Running debug with helm doesn't apply CRD changes:
helm3 upgrade --install dapr dapr/dapr --version=1.4.3 --namespace dapr-system --dry-run --debug

More info: https://github.com/dapr/dapr/issues/3731#issuecomment-941191734


`time="2021-10-11T12:18:35.519609747Z" level=info msg="log level set to: info" instance=dapr-operator-9bc45d8fc-czqd9 scope=dapr.operator type=log ver=unknown
time="2021-10-11T12:18:35.519784194Z" level=info msg="metrics server started on :9090/" instance=dapr-operator-9bc45d8fc-czqd9 scope=dapr.metrics type=log ver=unknown
time="2021-10-11T12:18:35.520200483Z" level=info msg="starting Dapr Operator -- version 1.4.3 -- commit a8ee30180e1183e2a2e4d00c283448af6d73d0d0" instance=dapr-operator-9bc45d8fc-czqd9 scope=dapr.operator type=log ver=unknown
I1011 12:18:36.573546       1 request.go:655] Throttling request took 1.044726144s, request: GET:https://172.20.0.1:443/apis/coordination.k8s.io/v1?timeout=32s
time="2021-10-11T12:18:36.73283813Z" level=info msg="Dapr Operator is started" instance=dapr-operator-9bc45d8fc-czqd9 scope=dapr.operator type=log ver=unknown
I1011 12:18:36.732953       1 leaderelection.go:243] attempting to acquire leader lease dapr-system/operator.dapr.io...
time="2021-10-11T12:18:36.940709962Z" level=info msg="getting tls certificates" instance=dapr-operator-9bc45d8fc-czqd9 scope=dapr.operator type=log ver=unknown
time="2021-10-11T12:18:36.940842904Z" level=info msg="tls certificates loaded successfully" instance=dapr-operator-9bc45d8fc-czqd9 scope=dapr.operator type=log ver=unknown
time="2021-10-11T12:18:36.941303961Z" level=info msg="starting gRPC server" instance=dapr-operator-9bc45d8fc-czqd9 scope=dapr.operator.api type=log ver=unknown
time="2021-10-11T12:18:36.941386697Z" level=info msg="Healthz server is listening on :8080" instance=dapr-operator-9bc45d8fc-czqd9 scope=dapr.operator type=log ver=unknown
time="2021-10-11T12:18:37.944396942Z" level=info msg="starting webhooks" instance=dapr-operator-9bc45d8fc-czqd9 scope=dapr.operator type=log ver=unknown
I1011 12:18:37.944450       1 leaderelection.go:243] attempting to acquire leader lease dapr-system/webhooks.dapr.io...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13e901a]
goroutine 436 [running[]:
github.com/dapr/dapr/pkg/operator.patchCRDs({0x18e43b0, 0xc0007bf7c0}, 0x0, {0xc000ade270, 0x1, 0x0})
    /home/runner/work/dapr/dapr/pkg/operator/webhooks.go:121 +0x4fa
created by github.com/dapr/dapr/pkg/operator.RunWebhooks
    /home/runner/work/dapr/dapr/pkg/operator/webhooks.go:68 +0x665`

I tried a downgrade to 1.4.2 and now I can't apply a Subscription:
`unable to recognize "": no matches for kind "Subscription" in version "dapr.io/v2alpha1"`

## Steps to Reproduce the Problem

Upgrade from 1.4.2 to 1.4.3

## Release Note

<!-- How should the fix for this issue be communicated in our release notes? It can be populated later. -->
<!-- Keep it as a single line. Examples: -->

<!-- RELEASE NOTE: **ADD** New feature in Dapr. -->
<!-- RELEASE NOTE: **FIX** Bug in runtime. -->
<!-- RELEASE NOTE: **UPDATE** Runtime dependency. -->

RELEASE NOTE:

--- a/pkg/operator/webhooks.go
+++ b/pkg/operator/webhooks.go
@@ -118,8 +118,11 @@ func patchCRDs(ctx context.Context, conf *rest.Config, crdNames ...string) {
 			continue
 		}
 
-		if crd.Spec.Conversion.Webhook.ClientConfig == nil {
-			log.Errorf("CRD %q does not have an existing webhook client config", crdName)
+		if crd == nil ||
+			crd.Spec.Conversion == nil ||
+			crd.Spec.Conversion.Webhook == nil ||
+			crd.Spec.Conversion.Webhook.ClientConfig == nil {
+			log.Errorf("CRD %q does not have an existing webhook client config. Applying resources of this type will fail.", crdName)
 
 			continue
 		}

